src/api/java/com/minecolonies/api/tileentities/AbstractTileEntityGrave.java

```diff
@@ -0,0 +1,192 @@
+package com.minecolonies.api.tileentities;
+
+import com.minecolonies.api.colony.ICitizenData;
+import net.minecraft.inventory.container.INamedContainerProvider;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.tileentity.ITickableTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nonnull;
+
+import static com.minecolonies.api.research.util.ResearchConstants.RESURECT_CHANCE;
+import static com.minecolonies.api.util.constant.Constants.*;
+
+public abstract class AbstractTileEntityGrave extends TileEntity implements INamedContainerProvider, ITickableTileEntity
+{
+    /**
+     * The inventory of the tileEntity.
+     */
+    protected ItemStackHandler inventory;
```

Comment:
maybe for the inventory we could just extend our rack inventories. So we can query them efficiently too

Reply:
I wouldnt make this tickable either, just let the gravediggers building or worker handle the decay

Reply:
This would make the grave dependent on the graveyard building. In my design, the grave spawn even if you don't have a graveyard to allow the player to retrieve the items manually.

Reply:
Yeah this is fine.
Performance wise it does not matter all that much and this provides more flexibility to the player.